tests/debugpy/test_run.py

```diff
@@ -284,3 +282,38 @@ def code_to_debug():
 
         using_frame_eval = backchannel.receive()
         assert using_frame_eval == (frame_eval == "yes")
+
+
+@pytest.mark.skipif(
+    not sys.version_info[0] >= 3,
+    reason="Test structure must still be updated to properly support Python 2 with unicode",
+)
+@pytest.mark.parametrize("run", [runners.all_launch[0]])
+def test_unicode_dir(tmpdir, run, target):
```

Comment:
Note: this works on Python 3 but fails on Python 2... in the tests there seems to be quite a bit of unicode/bytes mixing and it'll probably take a good while to sort it out, so, I decided to provide the pull request as is for now and leave the test only for Python 3.

I manually tested that it's possible to debug a file with unicode chars -- as long as they're compatible with the filesystem (unfortunately this means that we could break this again in the future, but given that Python 2 is becoming less relevant now that it's officially unsupported I'm not sure it's worth it doing the revamp on the test structure to stop mixing bytes and unicode in it).

Reply:
Yeah, I don't think we can run such a test reliably on Windows anyway, since it would depend on user-global locale to be set to something that can actually round-trip this path. It would need a container that the test suite can fully isolate and control. This might come anyway for other reasons eventually, but it's definitely not worthwhile for a py2-only corner case.